client/blocks/subscription-length-picker/index.jsx

```diff
@@ -106,20 +138,26 @@ export class SubscriptionLengthPicker extends React.Component {
 	};
 }
 
-export function myFormatCurrency( price, code ) {
+export function myFormatCurrency( price, code, options = {} ) {
 	const precision = CURRENCIES[ code ].precision;
 	const EPSILON = Math.pow( 10, -precision ) - 0.000000001;
 
 	const hasCents = Math.abs( price % 1 ) >= EPSILON;
-	return formatCurrency( price, code, hasCents ? {} : { precision: 0 } );
+	return formatCurrency( price, code, hasCents ? options : { ...options, precision: 0 } );
```

Comment:
It's a new behavior, so let's [extend its test suite](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/subscription-length-picker/test/index.js#L81) to cover it.